minionsai/discriminator_only/translator.py

```diff
@@ -57,30 +85,122 @@ def translate(self, game: Game):
         opp_money = min(all_money[1 - game.active_player_color], self.MAX_MONEY)
         score_diff = max(min(scores[game.active_player_color] - scores[game.inactive_player_color], self.MAX_SCORE_DIFF), -self.MAX_SCORE_DIFF) + self.MAX_SCORE_DIFF
         
-        return {
-            'board': np.array([board_obs]),  # shape is [batch, num_items=BOARD_SIZE^2]
-            'remaining_turns': np.array([[remaining_turns]]),  # shape is [batch, num_items=1]
-            'money': np.array([[money]]),
-            'opp_money': np.array([[opp_money]]),
-            'score_diff': np.array([[score_diff]])
-        }
+        d = OrderedDict()
+        d['board'] = np.array([board_obs])  # shape is [batch, num_items=BOARD_SIZE^2]
+        d['money'] = np.array([[money]])
+        d['remaining_turns'] = np.array([[remaining_turns]])  # shape is [batch, num_items=1]
+        d['opp_money'] = np.array([[opp_money]])
+        d['score_diff'] = np.array([[score_diff]])
+        if self.mode == "generator":
+            d['phase'] = np.array([[phase]])
+            # legal_actions = [self.valid_actions(game).flatten()]
+            # d['legal_actions'] = np.array([])
+        return d
+
+
+
+    def get_possible_legal_actions(self):
+        possibly_legal = np.zeros((self.num_things, self.num_things), bool)
+        for i in range(BOARD_SIZE ** 2 + 1):
+            for j in range(BOARD_SIZE ** 2 + 1):
+                # money -> board = spawn
+                # money -> money = advance phase
+                if i == self.spawn_pos:
+                    possibly_legal[i,j] = True
+                else:
+                    if j == self.spawn_pos:
+                        continue
+                    x1, y1 = convert_loc_to_pos(i)
+                    x2, y2 = convert_loc_to_pos(j)
+                    if abs(x1-x2) <= MAX_SPEED_OR_RANGE and abs(y1-y2) <= MAX_SPEED_OR_RANGE:
+                        possibly_legal[i,j] = True
+        return possibly_legal
+
+    def valid_actions(self, game: Game):
```

Comment:
It might be good to unit test these valid-action functions. I imagine if they have subtle edge-case errors we'll get horribly confused, and they're pretty self contained. e.g. you could assemble a random game state and just test that the valid_actions array is exactly what it should be.